### PR TITLE
Remove unused and undocumented Makefile variables.

### PR DESCRIPTION
**Pull request by themperek**

### CODE REVIEW

File: cocotb/share/makefiles/Makefile.sim [hunk 1]
```diff
@@ -130,8 +130,6 @@ CUSTOM_SIM_DEPS += $(shell find $(COCOTB_PY_DIR)/cocotb/ -name "*.py")
 # This triggers a recompilation of the simulation if cocotb library is updated.
 CUSTOM_SIM_DEPS += $(shell find $(LIB_DIR)/ -name "*")
 
-PWD := $(shell pwd)
```
Comment (eric-wieser) [inline cocotb/share/makefiles/Makefile.sim:133]: Are we sure all the other makefiles define this? It was added recently to fix a big where we forgot to define it I think

Comment (eric-wieser) [reply cocotb/share/makefiles/Makefile.sim:133]: #1300

Comment (themperek) [reply cocotb/share/makefiles/Makefile.sim:133]: It was used to set `PYTHONPATH` now is here: https://github.com/cocotb/cocotb/blob/master/cocotb/__init__.py#L63

Comment (eric-wieser) [reply cocotb/share/makefiles/Makefile.sim:133]: So to confirm, there are no makefiles left that use but do not define this variable?

Comment (themperek) [reply cocotb/share/makefiles/Makefile.sim:133]: Yes. To what I can see. CI should fail also.

Comment (eric-wieser) [reply cocotb/share/makefiles/Makefile.sim:133]: Here's one that seems undefined: https://github.com/cocotb/cocotb/blob/master/tests/designs/avalon_module/Makefile#L42

Comment (themperek) [reply cocotb/share/makefiles/Makefile.sim:133]: It is in CI so how does this work?

Comment (eric-wieser) [reply cocotb/share/makefiles/Makefile.sim:133]: I think that's the only one

Comment (themperek) [reply cocotb/share/makefiles/Makefile.sim:133]: HA: https://unix.stackexchange.com/questions/174990/what-is-pwd-vs-current-working-directory

Comment (themperek) [reply cocotb/share/makefiles/Makefile.sim:133]: And we do `SHELL := bash` also here: https://github.com/cocotb/cocotb/blob/master/cocotb/share/makefiles/Makefile.sim#L34

Comment (themperek) [reply cocotb/share/makefiles/Makefile.sim:133]: Should I remove all this `$(shell pwd)` and just use `$(PWD)` ? Separate PR?

Comment (eric-wieser) [reply cocotb/share/makefiles/Makefile.sim:133]: I don't think we should do that, apparently the variable is empty within sudo

Comment (eric-wieser) [reply cocotb/share/makefiles/Makefile.sim:133]: Let's worry about that later, anyway

Comment (eric-wieser) [reply cocotb/share/makefiles/Makefile.sim:133]: But I think we should add the one missing call to that file, for consistency

Comment (themperek) [reply cocotb/share/makefiles/Makefile.sim:133]: BTW: https://pubs.opengroup.org/onlinepubs/009604599/utilities/xcu_chap02.html 
Maybe should even avoid `PWD:=$(shell pwd)`. Yes. For another time I guess.

Comment (eric-wieser) [reply cocotb/share/makefiles/Makefile.sim:133]: `$(CURDIR)` is possibly a better choice, for a follow-up PR.

Comment (themperek) [reply cocotb/share/makefiles/Makefile.sim:133]: >But I think we should add the one missing call to that file, for consistency

Should be done.

File: tests/designs/avalon_module/Makefile [hunk 2]
```diff
@@ -39,7 +39,7 @@ else
 
 TOPLEVEL := burst_read_master
 
-COCOTB?=$(PWD)/../../..
+COCOTB?=$(shell pwd)/../../..
```
Comment (eric-wieser) [inline tests/designs/avalon_module/Makefile:42]: Would prefer we stay consistent with all the other files

```suggestion
PWD=$(shell pwd)

COCOTB?=$(PWD)/../../..
```

That way we can do further cleanup with a find and replace

**Pull request merged**